operator-kustomize/deployment.yaml

```diff
@@ -21,7 +21,10 @@ spec:
           resources:
             requests:
               memory: 256Mi
-          securityContext:
```

Comment:
why is this changed? @nitisht

Reply:
The docker client that we are using `client.NewEnvClient()` needs access to `/var/run/docker.sock` to talk to the host's docker daemon. 

But a process running as non-root doesn't have access to `/var/run/docker.sock`

Reply:
So this means that our operator will be root? @nitisht

Reply:
Yes, AFAICS. This also adds an implicit requirement that the host where operator is running should have Docker installed (which may not be the case always). Ideally i'd like change to some other library that allows copying from a container without strict Docker dependency but I could not find one.

Reply:
@nitisht can operator deploy temporary container via the `kube-apiserver` instead of talking directly to the `docker.sock`, that way we can leverage on the RBAC, im looking more into https://github.com/moby/moby right now but having direct access to `/var/run/docker.sock` sounds very dangerous